src/ExtensionDefinitionInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Interop\Container;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * An extension optionally implements this interface to reflect it's dependencies.
+ */
+interface ExtensionDefinitionInterface extends ServiceDefinitionInterface
```

Comment:
So far, the things in FIG have had formal PHP interfaces. In this case, we are talking about a callable, which sorta has an interface already. It would be great if we could simply declare a callable type: without that, you'd have to necessarily use OOP to create a service definition, whereas most simple extensions and factories are just `fn(): new Service()`. 

Psalm has a way of doing this, but I don't know if a Psalm declaration counts as a standard in FIG.

Reply:
There is no agreed-upon standard, but it does look like at least PHPStan and Psalm have adopted the same syntax.

Even if people regard the doc-blocks as just documentation, I would be fine with using this syntax - if it happens to also work for static analysis with two different tools, that's just an added bonus, right? 🙂

I am still very open to changing this PR back to just one interface and some doc-blocks, but there are a lot of considerations to make that decision - see #62 where I've tried to pitch that discussion.

Reply:
I've added the types

https://github.com/mindplay-dk/service-provider/commit/657e24bf7bad714a1a024c861475e17fe8c82e6c